velox/functions/sparksql/tests/ArrayMinTest.cpp

```diff
@@ -0,0 +1,223 @@
+        {S("red shiny car ahead"), S("blue clear sky above")},
+        {std::nullopt,
+         S("blue clear sky above"),
+         S("yellow rose flowers"),
+         S("orange beautiful sunset")},
+        {},
+        {std::nullopt, std::nullopt},
+        {S("red shiny car ahead"),
+         S("purple is an elegant color"),
+         S("green plants make us happy")},
+    });
+    auto expected = makeNullableFlatVector<S>(
+        {S("blue clear sky above"),
+         S("blue clear sky above"),
+         std::nullopt,
+         std::nullopt,
+         S("green plants make us happy")});
+    testExpr<S>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testBoolNullable() {
+    auto arrayVector = makeNullableArrayVector<bool>(
+        {{true, false},
+         {true},
+         {false},
+         {},
+         {std::nullopt, std::nullopt},
+         {true, false, true, std::nullopt},
+         {std::nullopt, true, false, true},
+         {false, false, false},
+         {true, true, true}});
+
+    auto expected = makeNullableFlatVector<bool>(
+        {false,
+         true,
+         false,
+         std::nullopt,
+         std::nullopt,
+         false,
+         false,
+         false,
+         true});
+    testExpr<bool>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testBool() {
+    auto arrayVector = makeArrayVector<bool>(
+        {{true, false},
+         {true},
+         {false},
+         {},
+         {false, false, false},
+         {true, true, true}});
+
+    auto expected = makeNullableFlatVector<bool>(
+        {false, true, false, std::nullopt, false, true});
+    testExpr<bool>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  template <typename T>
+  void testNumNullable() {
+    constexpr T kNaN = std::numeric_limits<T>::quiet_NaN();
+    auto arrayVector = makeNullableArrayVector<T>(
+        {{-1.0, 0.0, 1.0, kNaN, 2.0, 3.0, 4.0},
+         {4.0, 3.0, 2.0, 1.0, std::nullopt, 0.0, -1.0, -2.0, kNaN},
+         {-5.0, -4.0, -3.0, -2.0, -1.0},
+         {101.0, 102.0, 103.0, 104.0, std::nullopt},
+         {std::nullopt, -1.0, -2.0, std::nullopt, -3.0, -4.0},
+         {},
+         {std::nullopt, std::nullopt},
+         {kNaN, std::nullopt, -1.0, kNaN, -2.0},
+         {std::nullopt, kNaN}});
+    auto expected = makeNullableFlatVector<T>(
+        {-1.0,
+         -2.0,
+         -5.0,
+         101.0,
+         -4.0,
+         std::nullopt,
+         std::nullopt,
+         -2.0,
+         kNaN});
+    testExpr<T>(expected, "array_min(C0)", {arrayVector});
+  }
+};
+
+TEST_F(ArrayMinTest, intArrays) {
+  testIntNullable<int8_t>();
+  testIntNullable<int16_t>();
+  testIntNullable<int32_t>();
+  testIntNullable<int64_t>();
+  testInt<int8_t>();
+  testInt<int16_t>();
+  testInt<int32_t>();
+  testInt<int64_t>();
+}
+
+TEST_F(ArrayMinTest, varcharArrays) {
+  testInLineVarcharNullable();
+  testVarcharNullable();
+}
+
+TEST_F(ArrayMinTest, boolArrays) {
```

Comment:
For readability, split this test into 2 and remove helper functions testBoolNullable and testBool.

```
TEST_F(ArrayMinTest, bool)
TEST_F(ArrayMinTest, boolWithNulls)
```

Test names do not need 'Arrays' suffix as it is implied.

Reply:
rewrite all tests with simple interface like `array_max`